### ISSUE TITLE
Missing labels in v2 TCP stats

### ISSUE THREAD

**Issue by bianpengyuan** [istio/istio#20802]
Here is the server side v2 tcp metrics from the master build:
```
istio_tcp_received_bytes_total{
app="svc00-0-5-0",
connection_security_policy="NONE",
destination_app="svc00-0-5-0",
destination_port="0",
destination_principal="unknown",
destination_service="unknown",
destination_service_name="unknown",
destination_service_namespace="service-graph00",
destination_version="v1",
destination_workload="svc00-0-5-0",
destination_workload_namespace="service-graph00",
instance="10.56.3.224:15090",
job="envoy-stats",
namespace="service-graph00",
permissive_response_code="none",
permissive_response_policyid="none",
pod_name="svc00-0-5-0-65785bc555-kdmn5",
pod_template_hash="65785bc555",
reporter="destination",
request_protocol="tcp",
response_code="0",
response_flags="-",
role="service",
security_istio_io_tlsMode="istio",
source_app="svc00-0-5",
source_principal="unknown",
source_version="v1",
source_workload="svc00-0-5",
source_workload_namespace="service-graph00",
version="v1"
}
```

destination service labels are missing. and mtls labels are also not correctly set.

**Comment by howardjohn** [comment]
Once we fix this we should remove the skip from https://github.com/istio/istio/pull/19898

**Comment by gargnupur** [comment]
Fixed in istio release-1.5. Verified using test
https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/21303/integ-telemetry-k8s-tests_istio_release-1.5/370
istio_tcp_connections_opened_total{app="mongodb", connection_security_policy="MUTUAL_TLS", destination_app="mongodb", destination_canonical_service="unknown", destination_port="0", destination_principal="spiffe://cluster.local/ns/istio-bookinfo-1-56118/sa/default", destination_service="mongodb.istio-bookinfo-1-56118.svc.cluster.local", destination_service_name="mongodb", destination_service_namespace="istio-bookinfo-1-56118", destination_version="v1", destination_workload="mongodb-v1", destination_workload_namespace="istio-bookinfo-1-56118", instance="10.244.0.25:15090", job="envoy-stats", namespace="istio-bookinfo-1-56118", pod_name="mongodb-v1-77b445658d-br2bj", pod_template_hash="77b445658d", reporter="destination", request_protocol="tcp", response_code="0", response_flags="-", security_istio_io_tlsMode="istio", service_istio_io_canonical_name="mongodb", service_istio_io_canonical_revision="v1", source_app="ratings", source_canonical_service="unknown", source_principal="spiffe://cluster.local/ns/istio-bookinfo-1-56118/sa/bookinfo-ratings-v2", source_version="v2", source_workload="ratings-v2", source_workload_namespace="istio-bookinfo-1-56118", version="v1"}

Not closing it yet, because fix hasn't made istio/master yet..

**Comment by howardjohn** [comment]
Fixed in masted now
